nautobot/virtualization/views.py

```diff
@@ -242,18 +242,21 @@ def post(self, request, *args, **kwargs):
 #
 
 
-class VirtualMachineListView(generic.ObjectListView):
-    queryset = VirtualMachine.objects.all()
-    filterset = filters.VirtualMachineFilterSet
-    filterset_form = forms.VirtualMachineFilterForm
-    table = tables.VirtualMachineDetailTable
-    template_name = "virtualization/virtualmachine_list.html"
-
-
-class VirtualMachineView(generic.ObjectView):
-    queryset = VirtualMachine.objects.select_related("software_version", "tenant__tenant_group")
+class VirtualMachineUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.VirtualMachineBulkEditForm
+    filterset_class = filters.VirtualMachineFilterSet
+    filterset_form_class = forms.VirtualMachineFilterForm
+    form_class = forms.VirtualMachineForm
+    serializer_class = serializers.VirtualMachineSerializer
+    table_class = tables.VirtualMachineDetailTable
+    queryset = VirtualMachine.objects.select_related(
```

Comment:
for core, still not clear to me where and what we should use select related on.

Reply:
Of the below, the only one I would think *might* need an explicit `select_related` would be `tenant__tenant_group` since the list view column is just `tenant` but its rendering can include the tenant group. I'd expect all of the others below to be automatic, but would appreciate testing to confirm that.